xskillscore/core/contingency.py

```diff
@@ -796,3 +796,222 @@ def _gerrity_s(table):
+    observations,
+    forecasts,
+    bin_edges,
+    dim=None,
+    drop_intermediate=True,
+    return_results="area",
+):
+    """Computes the relative operating characteristic for a range of thresholds.
+
+    Parameters
+    ----------
+    observations : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+        If bin_edges=='continuous', observations are binary.
+    forecasts : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+        If bin_edges=='continuous', forecasts are probabilities.
+    bin_edges : array_like, str
+        Bin edges for categorising observations and forecasts.
+        Bins include the left most edge, but not the right.
+        If bin_edges='continuous', calculate bin_edges from forecasts, equal to
+        sklearn.metrics.roc_curve(f_boolean, o_prob)
+    dim : str, list
+        The dimension(s) over which to compute the contingency table
+    drop_intermediate : bool, default=True
+        Whether to drop some suboptimal thresholds which would not appear
+        on a plotted ROC curve. This is useful in order to create lighter
+        ROC curves.
+    return_results: str, default='area'
+        Specify how return is structed:
+            * 'area': return only the area under the curve of ROC
+            * 'all_as_tuple': return hit and false alarm rate at each bin and area under
+                the curve of ROC as tuple
+            * 'all_as_metric_dim': return hit and false alarm rate at each bin and area
+                under the curve of ROC concatinated into new `metric` dimension
+
+    Returns
+    -------
+    xarray.Dataset or xarray.DataArray : reduced by dimensions ``dim``.
+        See ``return_results`` parameter.
+
+
+    Examples
+    --------
+    >>> f = xr.DataArray(
+    ...     np.random.normal(size=(1000)),
+            coords=[('time', np.arange(1000))]
+    ... )
+    >>> o = xr.DataArray(
+    ...    np.random.normal(size=(1000)),
+    ...    coords=[('time', np.arange(1000))]
+    ... )
+    >>> category_edges = np.linspace(-2,2,5)
+    >>> roc(o, f, category_edges, dim=['time'])
+    <xarray.DataArray 'histogram_observations_forecasts' ()>
+    array(0.46812223)
+
+    See also
+    --------
+    xskillscore.Contingency
+    sklearn.metrics.roc_curve
+
+    References
+    ----------
+    http://www.cawcr.gov.au/projects/verification/
+    """
+
+    if dim is None:
+        dim = list(forecasts.dims)
+    if isinstance(dim, str):
+        dim = [dim]
+
+    continuous = False
+    if isinstance(bin_edges, str):
+        if bin_edges == "continuous":
+            continuous = True
+            # check that o binary
+            if isinstance(observations, xr.Dataset):
+                o_check = observations.to_array()
+            else:
+                o_check = observations
+            if str(o_check.dtype) != "bool":
+                if not ((o_check == 0) | (o_check == 1)).all():
```

Comment:
No we don’t want that...

Reply:
```python
o_check=observation_3d.to_array().chunk()

class CountingScheduler:
    """Simple dask scheduler counting the number of computes.
    Reference: https://stackoverflow.com/questions/53289286/"""

    def __init__(self, max_computes=0):
        self.total_computes = 0
        self.max_computes = max_computes

    def __call__(self, dsk, keys, **kwargs):
        self.total_computes += 1
        if self.total_computes > self.max_computes:
            raise RuntimeError(
                "Too many computes. Total: %d > max: %d."
                % (self.total_computes, self.max_computes)
            )
        return dask.get(dsk, keys, **kwargs)

def raise_if_dask_computes(max_computes=0):
    # return a dummy context manager so that this can be used for non-dask objects
    scheduler = CountingScheduler(max_computes)
    return dask.config.set(scheduler=scheduler)

with raise_if_dask_computes():
    ((o_check == 0) | (o_check == 1)).all()
```
doesnt raise

Reply:
I don't really understand what you're doing here. Can you use `dask.is_dask_collection()` and only check that data is binary if it is not a dask array?